speech_cleanup.qmd

```diff
@@ -0,0 +1,120 @@
+```{python}
+import pandas_gbq
+import os
+import pandas as pd
+import seaborn as sns
+import numpy as np
+import re
+```
+
+```{python}
+
+os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "token/gcp_token.json"
+
+df = pandas_gbq.read_gbq("""
+                    SELECT  *
+                    FROM `singapore-parliament-speeches.prod_mart.mart_speeches`
+                         WHERE member_name != ''
+                         AND member_name != 'Speaker'
+                         AND is_primary_question = false
+                    """)
+```
+
+```{python}
+[i for i in df.query('count_speeches_words<50').speech_text]
+
+```
+
+Some visuals, most speeches are within 2000 words with serious outliers
+
+```{python}
+sns.histplot(x = 'count_speeches_words', data = df)
+```
+
+What types of speeches?
+
+```{python}
+
+sns.histplot(y = 'topic_type_name', data = df)
+
+```
+
+What do short speeches look like?
+
+```{python}
+perc_10 = np.percentile(df.count_speeches_words, 10)
+[i for i in df.query(f'count_speeches_words<{perc_10}').speech_text]
+
+```
+
+What do different types of speeches look like?
+
+```{python}
+
+[i for i in df.query('topic_type_name.str.contains("Bill Introduced")').speech_text]
+
+[i for i in df.query('topic_type_name.str.contains("Correction by Written Statements")').speech_text]
+
+```
+
+Some patterns revealed
+1. Many procedural points - 'Yes please', 'Aye', 'last qns', etc. These are all very short and by speaker
+2. empty open brackets
+3. responses to speaker - 'thank you', etc.
+4. there are questions, end with ? and not classified as qns
+5. there are a number of vernacular speeches 
+6. We can omit bill introductions - these are procedural and serves merely to introduce the speech
```

Comment:
i looked at speeches which matched the criterion:
```python

```

and found that it would filter out speeches which i believe should be counted. for example, this speech should not be filtered out:

> "Mr Deputy Speaker, Sir, I will deliver the first part of my speech in Mandarin. (In Mandarin)  [Please refer to  Vernacular Speech.] I support the Environmental Public Health (Amendment) **Bill introduced** by the Ministry of Sustainability and the Environment. The ongoing global pandemic has given us a... (...)"




this was found using the query:
```sql
select
    topic_type,
    topic_title,
    speech_text,
    contains_substr(speech_text, 'bill introduced') as is_flagged
from `singapore-parliament-speeches.prod_mart.mart_speeches`
where contains_substr(speech_text, 'bill introduced') is true
```

Reply:
my proposal instead, would be to omit `topic_type = 'BI'` from the count of speeches. this is the topic where the bills are introduced, and there are rarely speeches made here.